src/drakvuf.cpp

```diff
@@ -161,6 +161,57 @@ int drakvuf_c::start_plugins(const bool* plugin_list, const plugins_options* opt
     return 1;
 }
 
+int drakvuf_c::stop_plugins(const bool* plugin_list)
+{
+    bool failed = false;
+    bool pending = false;
+
+    for (int i = 0; i < __DRAKVUF_PLUGIN_LIST_MAX; i++)
+    {
+        if (plugin_list[i])
+        {
+            int rc = plugins->stop(static_cast<drakvuf_plugin_t>(i));
+            if (rc < 0)
+                failed = true;
+            else if (rc > 0)
+                pending = true;
+        }
+    }
+
+    if (failed)
+        return -1;
+    else if (pending)
+        return 1;
+    else
+        return 0;
+}
+
+struct stop_plugins_data
+{
+    drakvuf_c* _drakvuf_c;
+    const bool* plugin_list;
+};
+
+static bool is_stopped(drakvuf_t drakvuf, void* data)
+{
+    auto d = (struct stop_plugins_data*)data;
+    auto rc = drakvuf_is_interrupted(drakvuf);
+    if (SIGDRAKVUFTIMEOUT == rc)
+        return rc;
```

Comment:
This conversion from interrupt signal to bool is confusing

Reply:
My fault! Thanks!